### DISCUSSION TITLE
Query profiling

### DISCUSSION THREAD

**Discussion by cylewitruk** [diesel-rs/diesel#3914] [Q&A]
I don't think there is since I haven't found any docs/discussions on it thus far, but is there by chance any way to profile queries? Trying to track down slow queries in sqlite and the available tooling for that is rather lacking :) Simply being able to log the actual query times would even be a big help, even if verbose.

**Comment by weiznich** [comment]
https://github.com/diesel-rs/diesel/pull/3864 introduced an API that would support that as well. This is not released in any diesel version yet, but might be included in future feature releases.

**Reply by cylewitruk** [reply]
Cool, will check it out, thanks :)

**Comment by lsunsi** [comment]


**Reply by weiznich** [reply]
@lsunsi Please don't ping me to ask questions that are already answered. This is highly discouraged.
